src/fides/api/api/v1/endpoints/admin.py

```diff
@@ -27,23 +28,28 @@ class DBActions(str, Enum):
     ],
     status_code=status.HTTP_200_OK,
 )
-async def db_action(action: DBActions) -> Dict:
+async def db_action(action: DBActions, revision: Optional[str] = "head") -> Dict:
     """
     Initiate one of the enumerated DBActions.
     """
-    action_text = "upgrade"
 
-    if action == DBActions.reset:
-        if not CONFIG.dev_mode:
-            raise HTTPException(
-                status_code=status.HTTP_501_NOT_IMPLEMENTED,
-                detail="Resetting the application database outside of dev_mode is not supported.",
-            )
+    if action == DBActions.downgrade:
+        downgrade_db(CONFIG.database.sync_database_uri, revision=revision)  # type: ignore[arg-type]
+        action_text = "downgrade"
+    else:
+        action_text = "upgrade"
 
-        reset_db(CONFIG.database.sync_database_uri)
-        action_text = "reset"
+        if action == DBActions.reset:
+            if not CONFIG.dev_mode:
+                raise HTTPException(
+                    status_code=status.HTTP_501_NOT_IMPLEMENTED,
+                    detail="Resetting the application database outside of dev_mode is not supported.",
+                )
 
-    await configure_db(CONFIG.database.sync_database_uri)
+            reset_db(CONFIG.database.sync_database_uri)
+            action_text = "reset"
```

Comment:
- If action is "reset" and revision is also in request body (say it's a lower downrev)
- reset_db is called, and then configure_db is called, this will result in a downgrade.

Looks like this was already the existing behavior.  If we are separating out "downgrading" should we not allow a revision to be supplied on reset?